src/Yaapii.Atoms/IO/Zip.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/*
+using java.io.ByteArrayInputStream;
+using java.io.ByteArrayOutputStream;
+using java.io.File;
+using java.io.FileInputStream;
+using java.io.InputStream;
+using java.nio.file.Path;
+using java.util.zip.ZipEntry;
+using java.util.zip.ZipOutputStream;
+*/
+using System.IO;
+using Yaapii.Atoms;
+using Yaapii.Atoms.IO;
+
+// <summary>
+/// Zip all Files in an Directory
+/// 
+/// </summary>
+public sealed class Zip : IInput
+{
+
+    private readonly System.IO.DirectoryInfo origin;
+
+    /// <summary>
+    /// Zip all Files in an Directory
+    /// not recursive
+    /// </summary>
+    /// <param name="origin"> the directory with the files to zip</param>
+    public Zip(System.IO.DirectoryInfo origin)
+    {
+        this.origin = origin;
+    }
+
+    /// <summary>
+    /// The Zipped Files as a Stream
+    /// </summary>
+    /// <returns></returns>
+    public Stream Stream()
+    {
+        MemoryStream memory = new MemoryStream();
+        using (var zip = new System.IO.Compression.ZipArchive(memory, System.IO.Compression.ZipArchiveMode.Create, true))
```

Comment:
Please use usings in class head instead of calling namespaces absolute, if it is not necessary.